hubgrep/lib/fetch_results.py

```diff
@@ -56,15 +58,20 @@ def fetch_concurrently(keywords, hosting_service_interfaces: List[HostingService
 
         results = []
         errors = []
-        for future in futures.as_completed(to_do):
-            success, base_url, _results = future.result(timeout=5)  # TODO tweak this for better UX as it blocks pageload on broken external services
-            if success:
-                if _results:
-                    _normalize(_results)
-                    results += _results
-            else:
-                errors.append((base_url, _results))
-        
+
+        future_timeout = app.config['HOSTER_REQUESTS_TIMEOUT'] + 2
```

Comment:
why +2?

Reply:
it looks like you already have a default of 2 in HostingServiceInterface params

Reply:
this is what comes "after" the requests. the requests should time out after HOSTER_REQUESTS_TIMEOUT, but if they didnt (for whatever reason), the future will raise a timeout after HOSTER_REQUESTS... +2